feature/gribi/ate_tests/base_hierarchical_route_installation_test/README.md

```diff
@@ -0,0 +1,89 @@
+# TE-3.1: Base Hierarchical Route Installation
+
+## Summary
+Validate IPv4 AFT support in gRIBI with recursion.
+
+## Procedure
+* Connect ATE port-1 to DUT port-1 and ATE port-2 to DUT port-2.
+* Establish gRIBI client connection with DUT using default parameters.
+* Using gRIBI Modify RPC install the following IPv4Entry set:
+  * 198.51.100.0/24 to NextHopGroup containing one NextHop, specified to be 203.0.113.1/32
+  * 203.0.113.1/32 to NextHopGroup containing one NextHop specified to be the address of ATE port-2.
+* Forward packets between ATE port-1 and ATE port-2 (destined to 198.51.100.0/24)
+  and determine that packets are forwarded successfully.
+* Validate that both routes are shown as installed via AFT telemetry.
+* Ensure that removing the NextHopGroup containing 203.0.113.1/32 with a DELETE
+  operation results in traffic loss, and removal from AFT.
+* Repeat above test with the following (table) cases:
+  * Explicitly specified egress interface of DUT port-2.
+  * Explicitly specified MAC address for ATE port-2.
+  * Entry installed in a non-default network instance (with ATE port-1 and port-2 assigned to a non-default NI)
+* Ensure that the following cases result in an error being returned:
+  * Invalid IPv4 prefix in the IPv4Entry
+  * Missing NextHopGroup within an IPv4Entry
+  * Empty NextHopGroup
+  * Empty NextHop
+  * Invalid IPv4 address in NextHop.
+* Validate that REPLACE operations:
+  * Fail when using 2.0.0.0/8 within the Ipv4EntryKey (does not exist).
+  * After installing a second next-hop-group with a different ID, validate that
+    a REPLACE for 1.0.0.0/8 can update (in-place) the NHG to the new value.
+    Validate via traffic and telemetry.
+
+## Config Parameter coverage
+No configuration relevant.
+
+## Telemetry Parameter coverage
+
+For prefix:\
```

Comment:
The lines in this section shouldn't end with a `\`.  Most other test docs are using bulleted lists for paths here, i.e.

```
Parameters:
*   ipv4-unicast/ipv4-entry/state
*   ipv4-unicast/ipv4-entry/state/next-hop-group
```

Reply:
Done!